devito/function_manager.py

```diff
@@ -45,10 +45,20 @@ def generate_function_signature(self, function_descriptor):
             function_params = function_params + [param_vec_def]
         if self.mic_flag:
             function_params += [cgen.Pointer(cgen.POD(type_label, name+"_pointer")) for type_label, name in function_descriptor.value_params]
-            return cgen.FunctionDeclaration(cgen.Value(self._pymic_attribute + '\nint', function_descriptor.name),
+
+            # sets function return type if specified
+            return_type = function_descriptor.return_type if function_descriptor.return_type else self._pymic_attribute + '\nint'
+
+            return cgen.FunctionDeclaration(cgen.Value(return_type, function_descriptor.name),
                                             function_params)
         else:
             function_params += [cgen.POD(type_label, name) for type_label, name in function_descriptor.value_params]
+
+            # used in auto tune
+            if function_descriptor.return_type:  # sets return type if specified
+                return cgen.FunctionDeclaration(cgen.Value(function_descriptor.return_type, function_descriptor.name),
+                                                function_params)
+
             return cgen.Extern("C", cgen.FunctionDeclaration(cgen.Value('int', function_descriptor.name), function_params))
```

Comment:
Instead of this `if-else`, setting the default value of `function_descriptor.return_type` to `int` will achieve the same effect and you can remove this special case `if-else` block from here.